tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts

```diff
@@ -0,0 +1,196 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {scaleLinear, scaleLog} from 'd3';
+
+import {ScaleType} from './scale_types';
+
+export interface Scale {
+  forward(domain: [number, number], range: [number, number], x: number): number;
+
+  reverse(domain: [number, number], range: [number, number], x: number): number;
+
+  nice(minAndMax: [number, number]): [number, number];
```

Comment:
Okay, so I’m reading this as: “A `Scale` takes a `domain` and a `range`
(real intervals) and returns an iso between them. This is uncurried to
reduce allocations, for performance reasons.”

So far, so good. This immediately makes `forward` and `reverse` clear
(though nit: maybe call the parameter to `reverse` `y` rather than `x`).
But what is `nice`? Doubting [Nice][fr] or [`nice`][unix], but I can’t
guess from the signature or the implementations.

[fr]: https://en.wikipedia.org/wiki/Nice
[unix]: https://en.wikipedia.org/wiki/Nice_(Unix)

Reply:
Changed the name and left some comment about how it relates to `d3.nice`.